mavenlint.gemspec

```diff
@@ -14,5 +14,5 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'rspec', '3.12'
   s.add_development_dependency 'rubocop', '1.31'
 
-  s.required_ruby_version = '>= 2.7.5'
+  s.required_ruby_version = '>= 3.0.0'
```

Comment:
```suggestion
  s.required_ruby_version = '~> 3'
```
Having updated a lot of dependencies the last few weeks, I think we should use pessimistic locking for the dependencies of our gems. Ruby likely won't have breaking changes in 3.x.x, but in a major release they might release breaking changes again. We have it green right now. In five years when someone comes back in and works on it, we'd like for them to be able to start from green.

Reply:
`~> 3` will do the same as `>= 3.0.0` here, if we want to be pessimistic we need to make greater than 3 and lesser than 4. WDYT?

Reply:
> ~> 1.1 means that when you bundle install, you’ll get the highest-released gem version of thin between the range >= 1.1 and < 2.0.

The article doesn't mention the case we used, but I got information from the [source](https://bundler.io/guides/gemfile.html)

> ~> 0 is identical to >= 0.0 and < 1.0.

But, nonetheless, project [rubocop ruby check does not accept this pattern](https://github.com/mavenlink/mavenlint/actions/runs/7835062981/job/21379618282), it forces to use with at least the minor.

So, I changed to use the `~> 3.0`. WDYT?

Reply:
Yeah I think you're right. The `~> 3.0` should give us exactly what we want. Thanks for figuring that out.

![image](https://github.com/mavenlink/mavenlint/assets/1417659/3c586e46-a5ab-4e6e-b7c4-5f8e9323d845)